maintenance_equipment_hierarchy/models/maintenance_equipment.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2020 ForgeFlow S.L. (https://forgeflow.com)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models
+
+
+class MaintenanceEquipment(models.Model):
+
+    _inherit = "maintenance.equipment"
+    _parent_name = "parent_id"
+    _parent_store = True
+    _parent_order = "name"
+
+    parent_id = fields.Many2one(
+        "maintenance.equipment",
+        "Parent Equipment",
+        index=True,
+        ondelete="cascade",
+        track_visibility="onchange",
```

Comment:
This should be also changed in v13, as I've noticed:
```suggestion
        tracking=True,
```